evaluation-core/src/commonMain/kotlin/SemanticVersion.kt

```diff
@@ -39,7 +34,7 @@ internal data class SemanticVersion(
             if (version == null) {
                 return null
             }
-            val matchGroup = regex.matchEntire(version)?.groupValues ?: return null
+            val matchGroup = pattern.matchEntire(version)?.groupValues ?: return null
             val major = matchGroup[1].toIntOrNull() ?: return null
             val minor = matchGroup[2].toIntOrNull() ?: return null
             val patch = matchGroup[4].toIntOrNull() ?: 0
```

Comment:
Are the matchGroup indices still correct? Does it start from 0? What do index 0 and 3 represent?

Reply:
Yep!

![Screenshot 2024-07-23 at 11 29 51 PM](https://github.com/user-attachments/assets/b0230c9d-8395-4660-b1a9-376876f0f420)

0 is the full match
3 is everything after the 2nd `.`